### PR TITLE
Added dW/dt term to P_sol in post_processing 

### LINKED ISSUE #566: Add dW/dt term to P_sol in post_processing

### PR DESCRIPTION
**Pull request by harshtech123**
/closes #566 

The effective_loss is now defined as the sum of the original loss (Ploss) plus the new dW_th_dt term. This term ensures that the transient change in stored energy is included in the power balance.

### CODE REVIEW

**Comment by harshtech123** [comment]
@jcitrin please  can i have a quick review .

File: torax/post_processing.py [hunk 1]
```diff
@@ -213,8 +213,20 @@ def make_outputs(
...
+   # If a previous simulation state is available, compute the time derivative
+  # of the total stored thermal energy (dW_thermal/dt). Otherwise, default to zero.
+  if previous_sim_state is not None:
+      dW_th_dt = (W_thermal_tot - previous_sim_state.post_processed_outputs.W_thermal_tot) / sim_state.dt
+  else:
+      dW_th_dt = 0.0
+
+
+  # Add the dW/dt term to the power loss to account for non-stationary states.
+  effective_loss = Ploss + dW_th_dt
```
Comment (jcitrin) [inline torax/post_processing.py:226]: should be `Ploss - dW_th_dt` .

The intuition is that if the plasma is heating up, then its internal energy is increasing (positive dW_th_dt), and therefore this absorbed energy is _not_ being lost by the plasma, so `Ploss` has to become smaller.

File: torax/post_processing.py [hunk 2]
```diff
@@ -213,8 +213,20 @@ def make_outputs(
...
+
   # TODO(b/380848256): include dW/dt term
-  tauE = W_thermal_tot / Ploss
+  tauE = W_thermal_tot / effective_loss
```
Comment (jcitrin) [inline torax/post_processing.py:229]: Apologies about the misleading TODO. From conversations with experts, it turns out that H-factors (ratios of scaling law energy confinement times like tauH89, to the actual confinement time tauE) is often calculated in experiments with the uncompensated Ploss in tauE, due to the difficulty in ascertaining the necessary compensation factors in experimental databases.

There are other subtleties like whether to subtract (or not) the plasma radiation.

Therefore I recommend the following:

1. For now, remove this "effective loss" and keep tauE calculated as before.

2. Make `dW_th_dt` a new output in post_processed_outputs (need to extend `PostProcessedOutputs` in `state.py` , and then add `dW_th_dt` to updated_post_processed_outputs. It's anyway useful to have that output. I'll then explore what should be done with radiation + dW_th_dt for calculating different versions of tauE later on.

Comment (harshtech123) [reply torax/post_processing.py]: calculating tauE as before and added dW_th_dt a new output in post_processed_outputs

**Comment by harshtech123** [comment]
@jcitrin made reviewed changes , thank you !

**Review by jcitrin — APPROVED** [review]

**Comment by jcitrin** [comment]
Merged by https://github.com/google-deepmind/torax/commit/4b5b29433308547ed2a3bf6b312ae2b40725df1a

In addition, regenerated all the sim test data to include the new post_processed_output in the nc files
